src/ui/components/forms/VPNTextArea.qml

```diff
@@ -76,6 +78,28 @@ Item {
                 id: textAreaStates
                 itemToTarget: textArea
             }
+
+            function removeCharsInRange(string, startIndex, endIndex) {
+                return string.slice(0, startIndex) + string.slice(endIndex);
+            }
+
+            function handleOnTextChanged(text) {
+                const updatedTextLength = text.length;
+                const textInputLength = updatedTextLength - textLength;
+                const exceedsCharLimit = textLength + textInputLength > textArea.maxCharacterCount;
+
+                // Remove most recently added characters in case
+                // the text input exceeds the maximum length
+                if (exceedsCharLimit) {
```

Comment:
I think we can do this without having `textLength`. Let me try:

```
if (text.length > textArea.maxCharacterCount) {
    // The cursor is reset when `text` changes. Let's store the old value, normalized if needed.
    const cursorPos = textArea.cursorPosition > textArea.maxCharacterCount ? textArea.maxCharacterCount : textArea.cursorPosition;
    textArea.text = text.slice(0, textArea.maxCharacterCount);
    textArea.cursorPosition = cursorPos;
}
```

This should work.